lib/runtime/invalid_transaction_test.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package runtime
+
+import (
+	"testing"
+
+	"github.com/ChainSafe/gossamer/lib/transaction"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestInvalidTransactionErrors(t *testing.T) {
+	testCases := []struct {
+		name        string
+		test        []byte
+		expErr      error
+		expValidity *transaction.Validity
+	}{
+		{
+			name:   "ancient birth block",
+			test:   []byte{1, 0, 5},
+			expErr: errAncientBirthBlock,
+		},
+	}
+
+	for _, c := range testCases {
+		t.Run(c.name, func(t *testing.T) {
+			validity, err := UnmarshalTransactionValidity(c.test)
+			if c.expErr == nil {
+				require.NoError(t, err)
+			}
+
+			var valErr string
+			if err != nil {
+				valErr = err.Error()
+			}
+			require.Equal(t, c.expErr.Error(), valErr)
```

Comment:
See https://engineering-handbook-pr-62.onrender.com/development/tech-stack/go#testing-errors

So it's easier to change test cases in the future in case we wrap error context etc.

Reply:
Im getting error not in error chain when I do this